internal/provider/apikey_resource_test.go

```diff
@@ -0,0 +1,44 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"testing"
+)
+
+func createRandomApiKeyName() string {
+	return fmt.Sprintf("key-terraformprovider-%s", randomString())
+}
+
+func TestAccBasicApiKey(t *testing.T) {
+	apiKeyName := createRandomApiKeyName()
+	config := func(displayName string, ownerType string, ownerId string) string {
+		return fmt.Sprintf(`
+provider "temporalcloud" {
+
+}
+
+resource "temporalcloud_apikey" "terraform" {
+  display_name = "%s"
+	  owner_type = "%s"
+	  owner_id = "%s"
+	expiry_time = "2024-10-01T00:00:00Z"
+	description = "Test API Key"
+}`, displayName, ownerType, ownerId)
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			//{
+			//	Config: config(apiKeyName, "user", "user-id"),
+			//},
```

Comment:
I tried using "user" as owner type but I get back error "unsupported owner type: user". Based on the possible values in [[link](https://dotnet.temporal.io/api/Temporalio.Api.Cloud.Identity.V1.ApiKeySpec.html#Temporalio_Api_Cloud_Identity_V1_ApiKeySpec_OwnerType)] user should be acceptable.


"service-account" works and tests pass

Reply:
I believe this is because of the API Key in Temporal. 

- A user can create their own API Keys
-- A Global Admin can disable/enable and delete a user's API Key but not create one for the user
- A Global Admin can CRUD Service Account API Keys for the Service Accounts they created

Reply:
See [this](https://github.com/temporalio/api-cloud/blob/0a01b742b0aebd1e0cea0a48914f9b517e0ec174/temporal/api/cloud/cloudservice/v1/request_response.proto#L304
) comment in the protos 

The CreateAPIKey request is supported for Service Accounts only right now

Reply:
got it. So it only supports api keys for service-accounts! Though we might want to add CRUD for service accounts as well (might do it in another PR)

Reply:
Should we fail at the provider input level for now while validating? Otherwise, this will result in an apply error, which can be worse DX.